### PR TITLE
Fixes bug when iterating over empty list of links in extraction step of transcoding

### PR DESCRIPTION
**Pull request by alakra**

### CODE REVIEW

File: test/exhal/transcoder_test.exs [hunk 1]
```diff
@@ -79,6 +79,23 @@ defmodule ExHal.TranscoderTest do
...
+      use ExHal.Transcoder
+
+      deflink "up", param: :mylink
+    end
+
+    hal = """
+    {
+      "_links": {}
+    }
+    """
```
Comment (pezra) [inline test/exhal/transcoder_test.exs]: Please, add a test for `"up": []` and `"up": {"href": null}` too. Those are both legal, if awkward, formulations

File: lib/exhal/transcoder.ex [hunk 2]
```diff
@@ -202,9 +202,9 @@ defmodule ExHal.Transcoder do
 
     quote do
       def unquote(extractor_name)(doc, params) do
-        ExHal.get_links_lazy(doc, unquote(rel), fn -> nil end)
+        ExHal.get_links_lazy(doc, unquote(rel), fn -> [%{}] end)
         |> List.first
-        |> Map.fetch!(:href)
+        |> Map.get_lazy(:href, fn -> nil end)
```
Comment (pezra) [inline lib/exhal/transcoder.ex]: Would it be better to wait until the very end to do the `first`?
```elixir
ExHal.get_links_lazy(doc, unquote(rel), fn -> [] end)
|> Enum.map(&Map.get(&1, :href))
|> List.first
```

Comment (lmarlow) [inline lib/exhal/transcoder.ex]: `fn -> [] end` can be written as `&[]`

File: test/exhal/transcoder_test.exs (hunk 1 above)
Comment (alakra) [inline test/exhal/transcoder_test.exs]: Sure thing!

**Comment by pezra** [comment]
Merged. Thanks!
